authors/apps/articles/views.py

```diff
@@ -88,3 +94,28 @@ def delete(self, request, slug):
         return Response({
             'errors': 'sorry article with that slug doesnot exist'
         }, status=status.HTTP_404_NOT_FOUND)
+
+
+class ToggleFavoriteAPIView(generics.GenericAPIView):
+    """
+    ToggleFavoriteAPIView favorites and unfavorites an article
+    """
+    permission_classes = (permissions.IsAuthenticated,)
+    serializer_class = serializers.ArticleSerializer
+    renderer_classes = (ArticleJSONRenderer,)
+
+    def get_object(self, slug):
```

Comment:
this is going to bring about duplication, @FahdJamy defined the same function in one of his views. Please revise.